bukkit/src/main/java/org/wensheng/juicyraspberrypie/RemoteSession.java

```diff
@@ -429,11 +448,73 @@ private void handleEntityCommand(String c, String[] args, boolean entityIsPlayer
+                    mobGoals.addGoal(mob, Integer.MIN_VALUE, new EmptyGoal(plugin, "noMoveAI", GoalType.MOVE));
+                    mobGoals.addGoal(mob, Integer.MIN_VALUE, new EmptyGoal(plugin, "noTargetAI", GoalType.TARGET));
+                    mobGoals.addGoal(mob, Integer.MIN_VALUE, new EmptyGoal(plugin, "noJumpAI", GoalType.JUMP));
+                    mobGoals.addGoal(mob, Integer.MIN_VALUE, new EmptyGoal(plugin, "noLookAI", GoalType.LOOK));
+                    mobGoals.addGoal(mob, Integer.MIN_VALUE, new EmptyGoal(plugin, "noUnknownBehaviorAI", GoalType.UNKNOWN_BEHAVIOR));
+                }
+                send("ok");
+                break;
+            }
+            case "disableControl": {
+                if (entity instanceof Mob mob) {
+                    final MobGoals mobGoals = Bukkit.getMobGoals();
+                    mobGoals.removeGoal(mob, GoalKey.of(Mob.class, new NamespacedKey(plugin, "noMoveAI")));
+                    mobGoals.removeGoal(mob, GoalKey.of(Mob.class, new NamespacedKey(plugin, "noTargetAI")));
+                    mobGoals.removeGoal(mob, GoalKey.of(Mob.class, new NamespacedKey(plugin, "noJumpAI")));
+                    mobGoals.removeGoal(mob, GoalKey.of(Mob.class, new NamespacedKey(plugin, "noLookAI")));
+                    mobGoals.removeGoal(mob, GoalKey.of(Mob.class, new NamespacedKey(plugin, "noUnknownBehaviorAI")));
+                    final PersistentDataContainer persistentDataContainer = mob.getPersistentDataContainer();
+                    final boolean noAI = persistentDataContainer.has(new NamespacedKey(plugin, "noAI"), PersistentDataType.BOOLEAN);
+                    if (noAI) {
+                        persistentDataContainer.remove(new NamespacedKey(plugin, "noAI"));
+                        mob.setAI(false);
+                    }
+                }
+                send("ok");
+                break;
+            }
+            case "walkTo": {
+                if (entity instanceof Mob mob) {
+                    String x = args[0], y = args[1], z = args[2];
+                    if (args.length > 3) {
+                        x = args[1];
+                        y = args[2];
+                        z = args[3];
+                    }
+                    final Location loc = parseLocation(x, y, z);
+                    mob.getPathfinder().moveTo(loc);
+                }
+                send("ok");
                 break;
             }
             case "getDirection":
                 send(entity.getLocation().getDirection().toString());
                 break;
+            case "setDirection": {
+                String x = args[0], y = args[1], z = args[2];
+                if (args.length > 3) {
+                    x = args[1];
+                    y = args[2];
+                    z = args[3];
+                }
+                final Vector vector = new Vector(Double.parseDouble(x), Double.parseDouble(y), Double.parseDouble(z));

```

Comment:
This looks a lot like the `parseLocation()` method below, so this could be extracted as a `parseVector()` method, too.